### PR TITLE
feat: full feature parity

### PR DESCRIPTION
**Pull request by peterdeme**
- Full feature parity

### CODE REVIEW

File: app.go [hunk 1]
```diff
@@ -134,6 +134,58 @@ func (c *Client) UpdateAppSettings(ctx context.Context, settings *AppSettings) (
...
+}
+
+type CheckSqsResponse struct {
+	Status string                 `json:"status"`
+	Error  string                 `json:"error"`
+	Data   map[string]interface{} `json:"data"`
+	Response
+}
+
+// Checks whether the AWS credentials are valid for the SQS queue access.
```
Comment (yaziine) [inline app.go:150]:
```suggestion
// CheckSQS checks whether the AWS credentials are valid for the SQS queue access.
```

File: app.go [hunk 2]
```diff
@@ -134,6 +134,58 @@ func (c *Client) UpdateAppSettings(ctx context.Context, settings *AppSettings) (
 	return &resp, err
 }
 
+type CheckSqsRequest struct {
```
Comment (yaziine) [inline app.go:137]: Let's make SQS capital letters.
```suggestion
type CheckSQSRequest struct {
```

Comment (yaziine) [reply app.go:137]: in all occurences

File: blocklist.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
+package stream_chat // nolint: golint
```
Comment (yaziine) [inline blocklist.go:1]: why nolint here?

File: channel_test.go [hunk 4]
```diff
@@ -78,6 +78,29 @@ func TestClient_CreateChannel(t *testing.T) {
...
 }
 
+func TestChannel_GetManyMessages(t *testing.T) {
+	ctx := context.Background()
+	c := initClient(t)
+	userA := randomUser(t, c)
+	userB := randomUser(t, c)
+
+	ch := initChannel(t, c, userA.ID, userB.ID)
+	resp1, err := c.CreateChannel(ctx, ch.Type, ch.ID, userA.ID, nil)
```
Comment (yaziine) [inline channel_test.go:88]: not sure to understand why are we calling CreateChannel here?

File: user.go [hunk 5]
```diff
@@ -190,6 +190,19 @@ func (c *Client) FlagUser(ctx context.Context, targetID, flaggedBy string) (*Res
...
 }
 
+type GuestUserResponse struct {
+	User        *User  `json:"user"`
+	AccessToken string `json:"access_token"`
+	Response
+}
+
+// SetGuestUser creates a new guest user.
+func (c *Client) SetGuestUser(ctx context.Context, user *User) (*GuestUserResponse, error) {
```
Comment (yaziine) [inline user.go:200]: Not keen with that naming. What about `CreateGuestUser` or `GetGuestUser` ? feel free to ignore me :)

**Comment by yaziine** [comment]
Looks good overall, just a few comments.

File: channel_test.go (hunk 4 above)
Comment (peterdeme) [reply channel_test.go:88]: no clue either 😅 fixed it

File: user.go (hunk 5 above)
Comment (peterdeme) [reply user.go:200]: ✅

File: blocklist.go (hunk 3 above)
Comment (peterdeme) [reply blocklist.go]: it was a copy-pasta 🍝

**Review by ferhatelmas — APPROVED** [review]
LGTM

**Pull request merged**